src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -70,9 +70,17 @@ internal GeneratorExecutionContext(Compilation compilation, ParseOptions parseOp
         /// Adds source code in the form of a <see cref="string"/> to the compilation.
         /// </summary>
         /// <param name="hintName">An identifier that can be used to reference this source text, must be unique within this generator</param>
-        /// <param name="source">The source code to be add to the compilation</param>
+        /// <param name="source">The source code to add to the compilation</param>
         public void AddSource(string hintName, string source) => AddSource(hintName, SourceText.From(source, Encoding.UTF8));
 
+        /// <summary>
+        /// Adds source code in the form of a <see cref="StringBuilder"/> to the compilation.
+        /// This builder must not be further modified: undefined behavior will result.
+        /// </summary>
+        /// <param name="hintName">An identifier that can be used to reference this source text, must be unique within this generator</param>
+        /// <param name="source">The source code to add to the compilation. This builder must not be further modified: undefined behavior will result.</param>
+        public void AddSource(string hintName, StringBuilder source) => AddSource(hintName, new StringBuilderText(source, Encoding.UTF8, SourceHashAlgorithm.Sha1));
```

Comment:
We cannot hard code `Sha1` in the code anymore. The only reason we can have `Sha1` at all is for compatibility reasons. Most uses of `Sha` in the code base you see exist for this reason. 

@chsienki short term we need to make sure that all of the `SourceText` we add here use `SourceHashAlgorithm.Sha256` (temporarily strong but sure to be a problem at some point down the road). Long term I think we need to feed through the hash algorithm specified on the command line to the `Compilation` such that files added by generators inherit this.

Reply:
@jaredpar, makes sense. I'll open an issue to track this I'm the generators code.

Reply:
I updated 1 of the cases in this PR.